x/tokenfactory/types/genesis.go

```diff
@@ -0,0 +1,45 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// this line is used by starport scaffolding # genesis/types/import
+
+// DefaultIndex is the default capability global index
+const DefaultIndex uint64 = 1
+
+// DefaultGenesis returns the default Capability genesis state
+func DefaultGenesis() *GenesisState {
+	return &GenesisState{
+		FactoryDenoms: []GenesisDenom{},
+	}
+}
+
+// Validate performs basic genesis state validation returning an error upon any
+// failure.
+func (gs GenesisState) Validate() error {
+	seenDenoms := map[string]bool{}
```

Comment:
nit: we can be slightly more performant here and use `struct{}`, but maybe that's overkill.

Reply:
Yeah, for validate genesis, I don't think it's a big deal